add-chain/go.mod

```diff
@@ -1,18 +1,16 @@
 module github.com/ethereum-optimism/superchain-registry/add-chain
 
-go 1.21
+go 1.22
 
-replace github.com/ethereum-optimism/superchain-registry/superchain => ../superchain
```

Comment:
Did you mean to remove this and other replacement directives? Can you explain the reasoning?

Reply:
Yes it was intentional. Just a cleanup. Here is the reasoning:

When you are using a go.work file, Go automatically resolves dependencies between the modules listed in the use block based on their local paths. This eliminates the need for the replace directives to point to local directories. The workspace mode will already know where the local modules are located, and it will use them instead of fetching those dependencies from a remote repository.

Reply:
Will revert that part of the change if its concerning. Main purpose of this pr is to bump the go version

Reply:
Reasoning makes sense, thanks.